.github/actions/deploy/action.yaml

```diff
@@ -51,9 +51,11 @@ runs:
     - name: Install operator
       shell: bash
       run: |-
-        set -exu;
+        set -ex;
         export OPERATOR_IMAGE_NAME=$(yq eval '."zenko-operator" | .sourceRegistry + "/" + .image' solution/deps.yaml)
-        export OPERATOR_IMAGE_TAG=$(yq eval '."zenko-operator".tag' solution/deps.yaml)
+        if [[ -z "${OPERATOR_IMAGE_TAG}" ]]; then
```

Comment:
should this not be specified by a "parameter" of the composite step instead?

Reply:
As this is get by `yq` from `deps.yaml` I think we can keep it here, because anyway I will get it the same way in the step..
I prefer keep it here instead of having to use the same bash script in multiple places...

This is also why I removed the `u` from `set -exu` to be able to check if `OPERATOR_IMAGE_TAG` is not set, and if that's the case, get it from `deps.yaml`
If i keep the `u` it will just say that `OPERATOR_IMAGE_TAG` is not set and will exit here...

Reply:
> As this is get by yq from deps.yaml I think we can keep it here, because anyway I will get it the same way in the composite step..

the point of having it as an explicit parameter is to make it clear that it is an input that may be provided : instead of passing it through some env variable (which may be missed), making it a parameter makes it obvious & documented that it can be set.
it does not really change the way the script is used : either the parameter is set (and its value can be used), or a default behavior can get the value from the `deps.yaml` file.

> This is also why I removed the u from set -exu to be able to check if OPERATOR_IMAGE_TAG is not set, and if that's the case, get it from deps.yaml. If i keep the u it will just say that OPERATOR_IMAGE_TAG is not set and will exit here...

`if [[ -z ${VARIABLE} ]]` is strange, `-z` is a basic test operator (used with -z).
But you have multiple option to get things to work:

* testing if the variable is set with advanced test syntax `if ! [[ -v VAR ]]`
* testing if the variable is set with the isset operator `if ! isset VAR`
* providing a default value in the variable expension `${MY_VARIABLE:-defaultValue}`, or possibly `if [ -z ${VAR:-} ]`